### PR TITLE
added VMStorageOsDisk check and test

### PR DESCRIPTION
**Pull request by avivgu**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/terraform/checks/resource/azure/VMStorageOsDisk.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceCheck
+
+
+class VMStorageOsDisk(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure that Virtual Machines use managed disks"
+        id = "CKV_AZURE_92"
+        supported_resources = ['azurerm_virtual_machine']
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/azure/VMStorageOsDisk.py:9]: @hadarpl This resource is deprecated
image

File: checkov/terraform/checks/resource/azure/VMStorageOsDisk.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
+        storage_os_disk = conf.get('storage_os_disk')
+        storage_data_disk = conf.get('storage_data_disk')
+        if storage_os_disk and 'vhd_uri' in storage_os_disk[0]:
+            return CheckResult.PASSED
+        if storage_data_disk and 'vhd_uri' in storage_data_disk[0]:
+            return CheckResult.PASSED
+        return CheckResult.FAILED
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/azure/VMStorageOsDisk.py:20]: @hadarpl I believe your logic is inverted, `vhd_uri` means this is an unmanaged disk:
image

**Review by nimrodkor — CHANGES_REQUESTED** [review]

File: checkov/terraform/checks/resource/azure/VMStorageOsDisk.py (hunk 1 above)
Comment (avivgu) [reply checkov/terraform/checks/resource/azure/VMStorageOsDisk.py:9]: so we don't want this check?

Comment (nimrodkor) [reply checkov/terraform/checks/resource/azure/VMStorageOsDisk.py:9]: We want it, just on different resource types. They are listed in the image I attached

Comment (avivgu) [reply checkov/terraform/checks/resource/azure/VMStorageOsDisk.py]: Done

File: checkov/terraform/checks/resource/azure/VMStorageOsDisk.py (hunk 2 above)
Comment (avivgu) [reply checkov/terraform/checks/resource/azure/VMStorageOsDisk.py]: Done

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**